### ISSUE TITLE
Toolbelt module not exported in index.d.ts

### ISSUE THREAD

**Issue by majcherpiotrek** [true-myth/true-myth#395]
I tried to update the library to v6 In my project, but the submodule imports like `import Maybe from "true-myth/maybe"` or `import Toolbelt from "true-myth/toolbelt"` are not working with my ts configuration. Here is my `ts-config` for reference:

```json
{
    "compilerOptions": {
        "outDir": "./dist",
        "noImplicitAny": true,
        "module": "es6",
        "moduleResolution": "node",
        "target": "es5",
        "jsx": "react",
        "allowJs": true,
        "declaration": true,
        "strict": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "skipLibCheck": true,
        "esModuleInterop": true,
        "lib": ["DOM", "ES5", "ES2015", "ES2015.Promise"]
    },
    "include": ["src/**/*"],
    "exclude": ["node_modules", "dist", "src/**/*.test.tsx", "src/**/*.test.ts"]
}
```

Importing the `Maybe` and `Result` modules directly like `import { Maybe, Result } from "true-myth"` is working fine, however I checked the `index.d.ts` file and it does not re-export the `Toolbelt` module:

```typescript
export { Maybe } from './maybe.js';
export { Result } from './result.js';
export { Unit } from './unit.js';
//# sourceMappingURL=index.d.ts.map
```

The bottom line is that I need to use the functions that were moved to the `toolbelt` module, it would be great if it was available as a direct import. If not, I'd appreciate a hint on how to make it possible to use the submodule imports

**Comment by chriskrycho** [comment]
Yikes! Thanks for filing this, we clearly have a bug! Those *should* be available from the nested imports, but it may be that you have to flip into the ESM-compatible `"nodenext"` or `"Node16"` modes in your `tsconfig.json`. I'll try to take a look back at exactly how those changed between v5 and v6 and why it isn't resolving!

**Comment by chriskrycho** [comment]
I've just tested locally and confirmed (and will add a note to the changelog and docs): to resolve correctly on v6, you need to update your `tsconfig.json` as described in the previous comment. This worked "correctly" without that on v5, even though we were only publishing ESM, because we used what I like to call *the Ember hack*: namely, we just copied all the `.d.ts` files into the root of the repository so resolution worked "as expected."

@bmakuh and I will chat sometime after he's back from vacation about whether and how we should make it possible to use the module in an interop mode, because this is the second report we've had that gets at some of the interop challenges for packages which opt into full ESM mode with TS 4.7+. 🤔

**Comment by chriskrycho** [comment]
Your suggestion of re-exporting is an interesting one—it should be possible to do that in a simple feature release, so that you can do `import { Toolbelt } from 'true-myth'` *without* migrating to the module design.

And: sorry about the difficulty here, as well as the delay in getting back to you!

**Comment by bmakuh** [comment]
I agree on the idea of re-exporting—that seems like a pretty straightforward move for the time being!

**Comment by chriskrycho** [comment]
Whoops, meant to create a PR, not push to `main`. 😬 Well, that commit landed. 😂 I will try to cut a release with it tomorrow (along with, hopefully, some improvements to the docs infrastructure and internal linking).
